### ISSUE TITLE
Help=>About shows transparent window w/o content (Linux: 1.7.0RC)

### ISSUE THREAD

**Issue by MarkBergman-cbica** [CBICA/CaPTk#524]
When CaPTk is first launched, the "About" box displays correctly.

After dismissing the box (without checking "Never show again"), then hovering over some menu items, then choosing "Help => About", the Help window is shown without any content.
<img width="1280" alt="Screenshot_20190522_184248" src="https://user-images.githubusercontent.com/36490035/58213843-9f9e6100-7cc1-11e9-92c9-5bbe2d47cc3f.png">

```
# rpm -qf /lib64/libQt5OpenGL.so.5
qt5-qtbase-gui-5.11.3-1.fc28.x86_64

```

**Comment by ashishsingh18** [comment]
I am unable to reproduce this on Ubuntu 18.04. The "About" window displays correctly. Tried it multiple times.

**Comment by ashishsingh18** [comment]
This seems to be an issue with Qtwebengine. Many people have reported similar issues:
https://bugreports.qt.io/browse/QTBUG-55466
https://stackoverflow.com/questions/38540943/qt-5-7-black-screen-on-windows
https://bugreports.qt.io/browse/QTBUG-56757

**Comment by AlexanderGetka-cbica** [comment]
I'm also unable to reproduce this on Ubuntu or Windows. The bugs earlier posted by Ashish imply that the QtWebEngine bug has been "fixed" but has been recurring over many Qt versions even into the 5.14 series.
It's particularly odd that it shows at startup but not later. Even so, I'm not sure this is something we can fix without being able to reproduce it.

**Comment by ashishsingh18** [comment]
Closing as unable to reproduce. Please re-open if it reappears.

**Comment by AlexanderGetka-cbica** [comment]
So I just did a double check on 1.8.0.Alpha on Windows and I now see this same issue (the About panel is black, rather than clear, when opened after the first time). Very odd. Re-opening and working on a fix for this.

**Comment by ashishsingh18** [comment]
> 
> 
> So I just did a double check on 1.8.0.Alpha on Windows and I now see this same issue (the About panel is black, rather than clear, when opened after the first time). Very odd. Re-opening and working on a fix for this.

This is working correctly for me on Windows. I always see the content and never see a transparent or blank window.

**Comment by AlexanderGetka-cbica** [comment]
Odd. I'm trying a fix that parents the WebEngineView correctly and refreshes it when the dialog is called up again. Since this feature acts differently in package mode I'm going to try in that configuration first, and if it solves the issue for me, maybe Mark can also take a look. If it doesn't work then I think this issue may be out of our hands.

**Comment by ashishsingh18** [comment]
when you call reload from showevent, also test if it behaves fine with zoom.

**Comment by AlexanderGetka-cbica** [comment]
I gave it a test from a package and it worked for me, including preserving the zoom level.

**Comment by MarkBergman-cbica** [comment]
Still an issue on 1.8.0Alpha2.

Launching CaPTk (initial automatic display of Help->About window) looks fine.

After closing the About window and Going to "Help -> Usage" I get an opaque white window (perhaps this is not a Qt issue, but a lack of Usage content?).
![CaPTk_opaque_Usage](https://user-images.githubusercontent.com/36490035/86298513-e8666000-bbcb-11ea-9757-dae1bb899358.png)

**Comment by sarthakpati** [comment]
Cannot replicate on Windows and Linux (Ubuntu 18.04) systems. Could this an issue with how QtWebEngine is interacting with your window manager, perhaps?

**Comment by AlexanderGetka-cbica** [comment]
Yeah, this happens if content can't be loaded. Perhaps the issue on my PC was due to something else, but I found that manipulating my paths fixed it...  @MarkBergman-cbica Can you post your log file from that session?

**Comment by MarkBergman-cbica** [comment]
Yep, the path for the help file looks suspicious.

```
2020:07:01,18:45:02;CaPTk;bergman;Unable to start help page, file '/tmp/.mount_captkLj6rGI/usr/share/doc/index.html' not found
2020:07:01,18:45:02;CaPTk;bergman;applying DARK theme
2020:07:01,18:45:02;CaPTk;bergman;file reading succeeded, qss file path = /tmp/.mount_captkLj6rGI/usr/bin/../data/../etc/captk.qss
2020:07:01,18:45:52;CaPTk;bergman;ApplicationPreferences::SerializePreferences status: NoError
2020:07:01,18:45:58;CaPTk;bergman;Unable to start help page, file '/tmp/.mount_captkyDDURR/usr/share/doc/index.html' not found
2020:07:01,18:45:58;CaPTk;bergman;applying DARK theme
2020:07:01,18:45:58;CaPTk;bergman;file reading succeeded, qss file path = /tmp/.mount_captkyDDURR/usr/bin/../data/../etc/captk.qss
2020:07:01,18:52:02;CaPTk;bergman;ApplicationPreferences::SerializePreferences status: NoError
```

**Comment by AlexanderGetka-cbica** [comment]
That's quite odd.... The file it should be looking for according to the fHelpTutorial class called by the About button should be: 
"${CaPTkDataDir}/../share/docs/1_credits.html" 
@CBICA/devteam can someone help me do a sanity check on this?

**Comment by AlexanderGetka-cbica** [comment]
Specifically I'm confused about why it's looking for index.html... "1_credits.html" is hardcoded in. The only thing I can think of is that trying to open a directory (rather than a file) with the WebEngineView attempts to load index.html from in that directory as a fallback.

**Comment by AlexanderGetka-cbica** [comment]
@MarkBergman-cbica When you reopened this issue, you mentioned that the initial on-startup About page display works. Then you mentioned that the Usage page display is blank. 

Two clarifying questions for you, then:
Does using Help->About menu bring up the About page correctly (with visible content) now, or is that still broken?
If both are broken, is the behavior of the Help->Usage menu any different from the behavior of the Help->About menu? 

The path to the content itself does look weird, though I think this is just part of the AppImage mounting.
I do have to wonder why, at least in my 1.8.0.Alpha AppImage on linux, we have both ${mount_point}/usr/docs/ and a ${mount_point}/usr/share/doc/ with very different files inside. From what I can tell ${mount_point}/usr/share/doc/ does not contain the index.html that the Usage dialog requests. However, ${mount_point}/usr/docs/ _does_ have that file. 
I only see this on Linux but haven't checked on mac.

On Windows, the ${captk_install_dir}/share/doc/ location contains all the files necessary for _both_ the About and Usage windows,  but this location isn't used in DEVELOPER_MODE builds (instead it tries to use " ${captk_executable_dir}/../docs/html/" which doesn't exist so far as I can tell) so it gives the same blank view. The usage menu works fine for me in PACKAGE_PROJECT mode on Windows.

To sum up: the required files aren't generally in place in developer-mode/debug builds on either Linux or Windows, but in package-mode they are in different locations per platform.
If anyone has any other thoughts or observations on this, let me know. Otherwise I'll push a fix ASAP that fixes this for package mode.

**Comment by sarthakpati** [comment]
> Otherwise I'll push a fix ASAP that fixes this for package mode.

What "fix"? The documentation **should** be picked up from the source directory in DEVELOPER_MODE and from the share location in PACKAGE_MODE. From what I can see, the packaging is working perfectly fine and there is some other issue at play, here.

**Comment by ashishsingh18** [comment]
This is quite likely a webengine + window manager issue. 

@MarkBergman-cbica What flavor of Linux is this on? Are you running the release or building from sources?

@AlexanderGetka-cbica I don't think it has anything to do with packaging.

**Comment by MarkBergman-cbica** [comment]
> @MarkBergman-cbica When you reopened this issue, you mentioned that the initial on-startup About page display works. Then you mentioned that the Usage page display is blank.
> 
> Two clarifying questions for you, then:
> Does using Help->About menu bring up the About page correctly (with visible content) now, or is that still broken?

Calling Help->About does show the correct page.

> The path to the content itself does look weird, though I think this is just part of the AppImage mounting.

Yes.


> I do have to wonder why, at least in my 1.8.0.Alpha AppImage on linux, we have both ${mount_point}/usr/docs/ and a ${mount_point}/usr/share/doc/ with very different files inside. From what I can tell ${mount_point}/usr/share/doc/ does not contain the index.html that the Usage dialog requests. However, ${mount_point}/usr/docs/ _does_ have that file.
> I only see this on Linux but haven't checked on mac.

```
[bergman@huajiao tmp]$ CaPTk/1.8.0.Alpha2/captk 
Checking for compatible OpenGL - this will happen only once.
WARNING: Trying to run CaPTk GUI using software rendering - this might not work on all systems and in those cases, only the CLI will be available.
QObject::connect: No such slot fTrainingSimulator::TrainTestRadioButtonChecked()
QObject::connect:  (receiver name: 'fTrainingSimulator')
QObject::connect: No such slot fTrainingSimulator::TrainRadioButtonChecked()
QObject::connect:  (receiver name: 'fTrainingSimulator')
QObject::connect: No such slot fTrainingSimulator::TestRadioButtonChecked()
QObject::connect:  (receiver name: 'fTrainingSimulator')
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 724, resource id: 38589341, major code: 40 (TranslateCoords), minor code: 0
QObject::connect: No such slot fMainWindow::CallPerfusionMeasuresCalculation(const double, const bool, const bool, const bool, const std::string, const std::string)
QObject::connect:  (sender name:   'fPerfusionEstimator')

^Z
[2]+  Stopped                 CaPTk/1.8.0.Alpha2/captk
[bergman@huajiao tmp]$ bg
[2]+ CaPTk/1.8.0.Alpha2/captk &
[bergman@huajiao tmp]$ mount
CaPTk-1.8.0.Alpha2.bin on /tmp/.mount_captkmfSvx8 type fuse.CaPTk-1.8.0.Alpha2.bin (ro,nosuid,nodev,relatime,user_id=500,group_id=500)
[bergman@huajiao tmp]$ ls /tmp/.mount_captkmfSvx8
AppRun  CaPTk.desktop  CaPTk.png  usr
[bergman@huajiao tmp]$ ls /tmp/.mount_captkmfSvx8/usr/
bin/          data/         docs/         etc/          lib/          libexec/      licenses/     plugins/      resources/    share/        translations/ 
[bergman@huajiao tmp]$ ls /tmp/.mount_captkmfSvx8/usr/share/
applications/           doc/                    featureExtractionBatch/ icons/                  
[bergman@huajiao tmp]$ ls /tmp/.mount_captkmfSvx8/usr/share/doc/
0_overview_resize.png            libcapnp-0.5.3/                  libmircommon7/                   libxau6/                         libxshmfence1/
0_tutorial_1_mainUI_resized.png  libcups2/                        libmircore1/                     libxcb-glx0/                     libxt6/
10_integration_resize.png        libdbus-1-3/                     libmirprotobuf3/                 libxcb-present0/                 libxtst6/
1_credits.html                   libffi6/                         libnettle6/                      libxcb-sync1/                    libxxf86vm1/
applications_resize.png          libgbm1/                         libnspr4/                        libxcb-xfixes0/                  modalities_resize.png
applicationsSeg_resize.png       libgcrypt20/                     libnss3-nssdb/                   libxcb-xkb1/                     pre-processing_resize.png
comparisonMode.png               libglib2.0-0/                    libpcre32-3/                     libxcomposite1/                  Science_breast.png
dock_comparison_resize.png       libgmp10/                        libpng12-0/                      libxcursor1/                     Science_EGFRvIII.png
drawing_resize.png               libgnutls30/                     libprotobuf-lite9v5/             libxdamage1/                     Science_jco.png
featurepanel_resize.png          libgssapi-krb5-2/                libselinux1/                     libxdmcp6/                       Science_Recurrence.png
functionalityRepresentation.png  libidn11/                        libsqlite3-0/                    libxext6/                        Science_Survival.png
imageLoading_resize.png          libk5crypto3/                    libsystemd0/                     libxfixes3/                      seeds_resize.png
libavahi-client3/                libkrb5-3/                       libtasn1-6/                      libxi6/                          sliders_resize.png
libavahi-common3/                libkrb5support0/                 libwayland-client0/              libxkbcommon0/                   
libboost-filesystem1.58.0/       liblzma5/                        libwayland-server0/              libxkbcommon-x11-0/              
libboost-system1.58.0/           libmirclient9/                   libx11-xcb1/                     libxrender1/                     
[bergman@huajiao tmp]$ ls /tmp/.mount_captkmfSvx8/usr/docs/
0_overview_resize.png                 doxygen.css                          graph_legend.html           ht_SpecialApps.html      Perfusion_Alignment.html       seg_Geodesic.html
0_tutorial_1_mainUI_resized.png       doxygen.png                          graph_legend.md5            ht_SpecialApps.js        Perfusion_Derivatives.html     seg_GeoTrain.html
10_integration_resize.png             drawing_resize.png                   gs_commandLineUsage.html    ht_utilities.html        phiEstimator.html              seg_SNAP.html
1_credits.html                        dynsections.js                       gs_components.html          imageLoading_resize.png  preprocessing_bias.html        sliders_resize.png
annotated.html                        FAQ.html                             gs_drawing.html             index.html               preprocessing_biasN4.html      splitbar.png
applications_resize.png               FeatureExtraction.html               gs_FAQ.html                 index.qhp                preprocessing_brats.html       survivalPredictor.html
applicationsSeg_resize.png            featurepanel_resize.png              gs_features.html            Installation.html        preprocessing_dcm2nii.html     sync_off.png
arrowdown.png                         files.html                           gs_imageLoading.html        jquery.js                preprocessing_histoMatch.html  sync_on.png
arrowright.png                        folderclosed.png                     gs_imageVisualization.html  libraPapers.html         preprocessing_reg.html         tab_a.png
bc_s.png                              folderopen.png                       gs_keyboard.html            LungCancer_SBRT.html     pre-processing_resize.png      tab_b.png
bdwn.png                              functionalityRepresentation.png      gs_preprocessing.html       modalities_resize.png    preprocessing_susan.html       tab_h.png
BreastCancer_breastSegmentation.html  Getting_Started.html                 gs_seedpoints.html          modules.html             preprocessing_zScoreNorm.html  tabs.css
BreastCancer_LIBRA.html               Getting_Started.js                   gs_segmentation.html        namespaces.html          recurrencePredictor.html       tab_s.png
BreastCancer_texture.html             Glioblastoma_Atlas.html              gs_specializedApps.html     nav_f.png                ReleaseNotes.html              Technical_Reference.html
CaPTk.TAGFILE                         Glioblastoma_Confetti.html           gs_supportedImages.html     nav_g.png                resize.js                      Technical_Reference.js
closed.png                            Glioblastoma_Directionality.html     gs_tabDocking.html          nav_h.png                Science_breast.png             Training_Module.html
comparisonMode.png                    Glioblastoma_EGFRvIII.html           gs_utilities.html           navtree.css              Science_EGFRvIII.png           tr_Apps.html
custom.css                            Glioblastoma_Molecular.html          How_To_Guides.html          navtreedata.js           Science.html                   tr_buildFromSource.html
custom_dark_theme.css                 Glioblastoma_PHI.html                How_To_Guides.js            navtreeindex0.js         Science_jco.png                tr_FeatureExtraction.html
debug.log                             Glioblastoma_Pseudoprogression.html  ht_FeatureExtraction.html   navtree.js               Science.js                     tr_forDeves.html
Diffusion_Derivatives.html            Glioblastoma_Recurrence.html         ht_Preprocessing.html       open.png                 Science_Recurrence.png         tr_integration.html
dock_comparison_resize.png            Glioblastoma_Survival.html           ht_Preprocessing.js         pages.html               Science_Survival.png
doc.png                               Glioblastoma_WhiteStripe.html        ht_Segmentation.html        PCA_Extraction.html      seeds_resize.png
Download.html                         graph_legend.dot                     ht_Segmentation.js          People.html              seg_DL.html
[bergman@huajiao tmp]$ cat ~/.CaPTk/1.8.0.Alpha2/2020-07-02.log 
2020:07:02,11:11:12;CaPTk;bergman;Unable to start help page, file '/tmp/.mount_captk50RPRB/usr/share/doc/index.html' not found
2020:07:02,11:11:12;CaPTk;bergman;applying DARK theme
2020:07:02,11:11:12;CaPTk;bergman;file reading succeeded, qss file path = /tmp/.mount_captk50RPRB/usr/bin/../data/../etc/captk.qss
2020:07:02,11:11:20;CaPTk;bergman;ApplicationPreferences::SerializePreferences status: NoError
2020:07:02,11:16:46;CaPTk;bergman;Unable to start help page, file '/tmp/.mount_captk1Lwak8/usr/share/doc/index.html' not found
2020:07:02,11:16:46;CaPTk;bergman;applying DARK theme
2020:07:02,11:16:46;CaPTk;bergman;file reading succeeded, qss file path = /tmp/.mount_captk1Lwak8/usr/bin/../data/../etc/captk.qss
2020:07:02,11:16:58;CaPTk;bergman;ftp://www.nitrc.org/home/groups/captk/downloads/SampleData_1.6.0/All.zip
2020:07:02,11:17:03;CaPTk;bergman;https://www.cbica.upenn.edu/CaPTk_Model_Library
2020:07:02,11:17:09;CaPTk;bergman;https://www.cbica.upenn.edu/CaPTk_Model_Library
2020:07:02,11:17:14;CaPTk;bergman;ftp://www.nitrc.org/home/groups/captk/downloads/SampleData_1.6.0/SurvivalPredictor.zip
2020:07:02,11:17:20;CaPTk;bergman;ApplicationPreferences::SerializePreferences status: NoError
2020:07:02,11:20:08;CaPTk;bergman;Unable to start help page, file '/tmp/.mount_captkwsvm9j/usr/share/doc/index.html' not found
2020:07:02,11:20:08;CaPTk;bergman;applying DARK theme
2020:07:02,11:20:08;CaPTk;bergman;file reading succeeded, qss file path = /tmp/.mount_captkwsvm9j/usr/bin/../data/../etc/captk.qss
2020:07:02,11:20:14;CaPTk;bergman;ftp://www.nitrc.org/home/groups/captk/downloads/SampleData_1.6.0/All.zip
2020:07:02,11:20:17;CaPTk;bergman;ftp://www.nitrc.org/home/groups/captk/downloads/SampleData_1.6.0/SurvivalPredictor.zip
2020:07:02,11:20:22;CaPTk;bergman;https://www.cbica.upenn.edu/CaPTk_Model_Library
2020:07:02,11:20:27;CaPTk;bergman;ApplicationPreferences::SerializePreferences status: NoError
2020:07:02,11:22:58;CaPTk;bergman;Unable to start help page, file '/tmp/.mount_captkmfSvx8/usr/share/doc/index.html' not found
2020:07:02,11:22:58;CaPTk;bergman;applying DARK theme
2020:07:02,11:22:58;CaPTk;bergman;file reading succeeded, qss file path = /tmp/.mount_captkmfSvx8/usr/bin/../data/../etc/captk.qss
2020:07:02,11:43:52;CaPTk;bergman;ApplicationPreferences::SerializePreferences status: NoError
[2]+  Done                    CaPTk/1.8.0.Alpha2/captk
[bergman@huajiao tmp]$ 
```


> 
> On Windows, the ${captk_install_dir}/share/doc/ location contains all the files necessary for _both_ the About and Usage windows, but this location isn't used in DEVELOPER_MODE builds (instead it tries to use " ${captk_executable_dir}/../docs/html/" which doesn't exist so far as I can tell) so it gives the same blank view. The usage menu works fine for me in PACKAGE_PROJECT mode on Windows.
> 
> To sum up: the required files aren't generally in place in developer-mode/debug builds on either Linux or Windows, but in package-mode they are in different locations per platform.
> If anyone has any other thoughts or observations on this, let me know. Otherwise I'll push a fix ASAP that fixes this for package mode.

**Comment by MarkBergman-cbica** [comment]
> This is quite likely a webengine + window manager issue.

Hmmm... the rendering looks fine, but the Usage content cannot be found.

> 
> @MarkBergman-cbica What flavor of Linux is this on? Are you running the release or building from sources?

I'm running the 1.8Alpha2 release from NITRC.
> 
> @AlexanderGetka-cbica I don't think it has anything to do with packaging.

**Comment by AlexanderGetka-cbica** [comment]
In file fHelpDialog.cpp lines  somewhere around 44:
`
if CAPTK_PACKAGE_PROJECT
  m_docDir = cbica::normPath(m_dataDir + "/../share/doc/");
`

This is what I mean. The fHelpDialog class, responsible for the Usage menu, explicitly looks for its help file in what resolves to:
 On Windows: ${captk_install_dir}/share/doc/
On Linux under the mount point: ${mount_point}/usr/share/doc/

Mark's "about" menu now works, it's just the "usage" menu that is throwing errors. You can see from up above that Mark's ${mount_point}/usr/share/doc/ doesn't contain index.html. Something is wrong 
Per the file, line ~52:
`m_helpFileFullPath = cbica::normPath(m_docDir + "/index.html");`

What I meant was, it would be easy enough to change this to look in the other location for now. But the captk-pkg script can just copy the doxygen files into /share/doc and it should be good.

**Comment by ashishsingh18** [comment]
@AlexanderGetka-cbica  That could be it. There is no index.html file in ~/usr/share/doc as per @MarkBergman-cbica's earlier comment.

The "About' menu was always working perhaps, because it looks for a different html file: 1_credits.html, which is present at ~/usr/share/doc

See the code below for 'About' - from file fhelpTutorial.cpp
```
  m_dataDir = getCaPTkDataDir();
#if CAPTK_PACKAGE_PROJECT
  m_docDir = cbica::normPath(m_dataDir + "/../share/doc/");
#else
  m_docDir = std::string(PROJECT_SOURCE_DIR) + "../docs/html/";
#endif
  m_helpFileFullPath = m_docDir + "/1_credits.html";
```
And see the code below from file fhelpDialog.cpp
```
  m_dataDir = getCaPTkDataDir();
#if CAPTK_PACKAGE_PROJECT
  m_docDir = cbica::normPath(m_dataDir + "/../share/doc/");
#else
#if WIN32
  m_docDir = cbica::normPath(captk_currentApplicationPath + "/../docs/html/");
#else
  m_docDir = cbica::normPath(captk_currentApplicationPath + "/../docs/html/");
#endif
#endif
  m_helpFileFullPath = cbica::normPath(m_docDir + "/index.html");
```

**Comment by ashishsingh18** [comment]
@AlexanderGetka-cbica Was this done?

#1204 seems a duplicate.

**Comment by MarkBergman-cbica** [comment]
In the message dated: Fri, 24 Jul 2020 07:49:19 -0700,
The pithy ruminations from Ashish Singh on 
[[External] Re: [CBICA/CaPTk] Help=>About shows transparent window w/o content (Linux: 1.7.0RC) (#524)] were:
=> @AlexanderGetka-cbica Was this done?
=> 
=> #1204 seems a duplicate. 

Issue #1204 is the same behavior I witnessed in #524, but the OS is different and #524 was originally about a different problem.

**Comment by AlexanderGetka-cbica** [comment]
I am going to push the fix for this today. It is good to have the diagnostic info from #1204 as well.

**Comment by AlexanderGetka-cbica** [comment]
@sarthakpati Hey, just a question regarding this commit: https://github.com/sarthakpati/CaPTk/commit/938a2a09cab76409be31ab657a47c5334000a8df
The lines you removed copied some docs into the wrong folder. But those files don't actually seem to be included in the Linux package at all without those lines. Does your commit title mean the docs-copying should be part of another script? Where should I include the docs-copying command if not captk-pkg?

**Comment by sarthakpati** [comment]
So, that line was not required as the HTML files are copies via CMake [[ref](https://github.com/CBICA/CaPTk/blob/master/CMakeLists.txt#L708)]. This was placing them on the top-level location of the `appdir` but this location is never referenced [[ref](https://github.com/CBICA/CaPTk/blob/c51676e4d40a2eee3ae624044ea59dcbb35cf194/src/view/gui/fHelpDialog.cpp#L44)] and [[ref](https://github.com/CBICA/CaPTk/blob/bdb669a60b5028c507f22df931d2ae631770eab4/src/view/gui/fHelpTutorial.cpp#L59)].
